platform/macos/src/MGLMapView.mm

```diff
@@ -1402,10 +1383,10 @@ - (void)handlePanGesture:(NSPanGestureRecognizer *)gestureRecognizer {
         _mbglMap->cancelTransitions();
 
         if (gestureRecognizer.state == NSGestureRecognizerStateBegan) {
-            _scaleAtBeginningOfGesture = _mbglMap->getScale();
+            _zoomAtBeginningOfGesture = _mbglMap->getZoom();
         } else if (gestureRecognizer.state == NSGestureRecognizerStateChanged) {
-            CGFloat newZoomLevel = log2f(_scaleAtBeginningOfGesture) - delta.y / 75;
-            [self scaleBy:powf(2, newZoomLevel) / _mbglMap->getScale() atPoint:startPoint animated:NO];
+            CGFloat newZoomLevel = _zoomAtBeginningOfGesture - delta.y / 75;
```

Comment:
Did you test Shift-dragging to zoom in and out? Does the behavior match the previous behavior? Subtracting this delta value from a zoom level would presumably have a different effect than subtracting the same value from a scale.